benchexec/test_runexecutor.py

```diff
@@ -1199,6 +1199,135 @@ def test_uptime_without_lxcfs(self):
+        if not container.get_fuse_overlayfs_executable():
+            self.skipTest("missing fuse-overlayfs")
+
+        # Check if COV_CORE_SOURCE environment variable is set and remove it.
+        # This is necessary because the coverage tool will not work in the nested runexec.
+        restore_env = False
+        if "COV_CORE_SOURCE" in os.environ:
+            restore_env = True
+            del os.environ["COV_CORE_SOURCE"]
+
+        with tempfile.TemporaryDirectory(prefix="BenchExec_test_") as temp_dir:
+            overlay_dir = os.path.join(temp_dir, "overlay")
+            os.makedirs(overlay_dir)
+            test_file = os.path.join(overlay_dir, "TEST_FILE")
+            output_dir = os.path.join(temp_dir, "output")
+            os.makedirs(output_dir)
+            mid_output_file = os.path.join(output_dir, "mid_output.log")
+            inner_output_file = os.path.join(output_dir, "inner_output.log")
+            with open(test_file, "w") as f:
+                f.write("TEST_TOKEN")
+                f.seek(0)
+
+            outer_cmd = [
+                "python3",
+                runexec,
+                "--full-access-dir",
+                "/",
+                "--overlay-dir",
+                overlay_dir,
+                "--full-access-dir",
+                output_dir,
+                "--hidden-dir",
+                "/tmp",
+                "--output",
+                mid_output_file,
+                "--",
+            ]
+            mid_cmd = [
+                "python3",
+                runexec,
+                "--full-access-dir",
+                "/",
+                "--overlay-dir",
+                overlay_dir,
+                "--full-access-dir",
+                output_dir,
+                "--hidden-dir",
+                "/tmp",
+                "--output",
+                inner_output_file,
+                "--",
+            ]
+            inner_cmd = [
+                "/bin/sh",
+                "-c",
+                f"if [ $({self.cat} {test_file}) != TEST_TOKEN ]; then exit 1; fi; {self.echo} TOKEN_CHANGED >{test_file}",
+            ]
+            combined_cmd = outer_cmd + mid_cmd + inner_cmd
+
+            self.setUp(
+                dir_modes={
+                    "/": containerexecutor.DIR_FULL_ACCESS,
+                    "/tmp": containerexecutor.DIR_HIDDEN,
+                    overlay_dir: containerexecutor.DIR_OVERLAY,
+                    output_dir: containerexecutor.DIR_FULL_ACCESS,
+                },
+            )
+            outer_result, outer_output = self.execute_run(*combined_cmd)
+            self.check_result_keys(outer_result, "returnvalue")
+            self.check_exitcode(
+                outer_result, 0, "exit code of outer runexec is not zero"
+            )
+            with open(mid_output_file, "rb") as mid_output_file:
+                self.assertIn(
+                    b"returnvalue=0", mid_output_file.read().strip().splitlines()
```

Comment:
```suggestion
            with open(mid_output_file, "r") as mid_output_file:
                self.assertIn(
                    "returnvalue=0", mid_output_file.read().strip().splitlines()
```
Here I see no reason to use raw binary mode. Same below, actually.